config.go

```diff
@@ -0,0 +1,162 @@
+
+	// Deprecated
+	ApkPath string `env:"apk_path"`
+}
+
+// validate validates the Configs.
+func (c Configs) validate() error {
+	if err := c.validateJSONKeyPath(); err != nil {
+		return err
+	}
+
+	if err := c.validateWhatsnewsDir(); err != nil {
+		return err
+	}
+
+	if err := c.validateMappingFile(); err != nil {
+		return err
+	}
+
+	return c.validateApps()
+}
+
+// validateJSONKeyPath validates if service_account_json_key_path input value exists if defined and has file:// URL scheme.
+func (c Configs) validateJSONKeyPath() error {
+	if !strings.HasPrefix(string(c.JSONKeyPath), "file://") {
+		return nil
+	}
+
+	pth := strings.TrimPrefix(string(c.JSONKeyPath), "file://")
+	if exist, err := pathutil.IsPathExists(pth); err != nil {
+		return fmt.Errorf("failed to check if json key path exist at: %s, error: %s", pth, err)
+	} else if !exist {
+		return errors.New("json key path not exist at: " + pth)
+	}
+	return nil
+}
+
+// validateWhatsnewsDir validates if whatsnews_dir input value exists if provided.
+func (c Configs) validateWhatsnewsDir() error {
+	if c.WhatsnewsDir == "" {
+		return nil
+	}
+
+	if exist, err := pathutil.IsDirExists(c.WhatsnewsDir); err != nil {
+		return fmt.Errorf("failed to check if what's new directory exist at: %s, error: %s", c.WhatsnewsDir, err)
+	} else if !exist {
+		return errors.New("what's new directory not exist at: " + c.WhatsnewsDir)
+	}
+	return nil
+}
+
+// validateMappingFile validates if mapping_file input value exists if provided.
+func (c Configs) validateMappingFile() error {
+	if c.MappingFile == "" {
+		return nil
+	}
+
+	if exist, err := pathutil.IsPathExists(c.MappingFile); err != nil {
+		return fmt.Errorf("Failed to check if mapping file exist at: %s, error: %s", c.MappingFile, err)
+	} else if !exist {
+		return errors.New("mapping file not exist at: " + c.MappingFile)
+	}
+	return nil
+}
+
+// Apps returns the app to deploy, by prefering .aab files.
+func (c Configs) Apps() []string {
+	if len(c.ApkPath) > 0 {
+		return strings.Split(c.ApkPath, "|")
+	}
+
+	apps := strings.Split(c.AppPath, "\n")
+	if len(apps) == 0 {
+		apps = strings.Split(c.AppPath, "|")
+	}
+
+	var apks, aabs []string
+	for _, pth := range apps {
+		pth = strings.TrimSpace(pth)
+		if filepath.Ext(pth) == ".aab" {
+			aabs = append(aabs, pth)
+		} else {
+			apks = append(apks, pth)
+		}
+	}
+
+	if len(aabs) > 0 && len(apks) > 0 {
+		log.Warnf("Both .aab and .apk files provided, using the .aab file(s): %s", strings.Join(aabs, ","))
+	}
+
+	if len(aabs) > 1 {
+		log.Warnf("More then 1 .aab files provided, using the first: %s", aabs[0])
+	}
+
+	if len(aabs) > 0 {
+		return []string{aabs[0]}
+	}
+
+	return apks
+}
+
+// validateApps validates if files provided via apk_path are existing files,
+// if apk_path is empty it validates if files provided via app_path input are existing .apk or .aab files.
+func (c Configs) validateApps() error {
```

Comment:
you have mostly the same parsing logic in Apps() method

Reply:
you are validating files that might will not even used, in case of .aab file the apks no need to be validated becuase they will not be used